org-roam-link.el

```diff
@@ -0,0 +1,341 @@
+               (string= "file" (org-element-property :type elem)))
+      (org-roam-link--convert-file-to-roam-link elem))))
+
+(defun org-roam-link-convert-buffer-links (file-or-roam)
+  "Convert all Org-roam links in the current buffer.
+FILE-OR-ROAM should be one of `file' or `roam', and
+indicate the link-type the buffer should convert to."
+  (interactive
+   (let ((completion-ignore-case t))
+     (list (completing-read "Convert to link type: " '("file" "roam") nil t))))
+  (save-excursion
+    (goto-char (point-min))
+    (org-next-link)
+    (while (not org-link--search-failed)
+      (let* ((elem (org-element-context))
+             (type (org-element-property :type elem)))
+        (if (and (string= file-or-roam "roam")
+                 (string= type "file"))
+            (org-roam-link--convert-file-to-roam-link elem)
+          (when (and (string= file-or-roam "file")
+                     (string= type "roam"))
+            (org-roam-link--convert-roam-to-file-link elem)))
+        (org-next-link)))))
+
+(defun org-roam-link--auto-create-roam-link-file (title &optional manual)
+  "Call `org-roam-capture' with a template using :immediate-finish t.
+TITLE is the title for the file to be created.
+MANUAL is boolean allowing manual selection of capture template(s)."
+  (let ((org-roam-capture--info (list (cons 'title title)
+                                      (cons 'slug (org-roam--title-to-slug title))))
+        (org-roam-capture--context 'title))
+    (if manual (org-roam-capture--capture)
+      (org-roam-capture--capture :keys "a"))))
+
+(defun org-roam-link-auto-create-roam-links-in-buffer (&optional manual)
+  "Create all non-existant roam-link files in current buffer.
+MANUAL is boolean which allows manual selection of `org-roam-capture'
+templates for each file created.
+If called with PREFIX `C-u' then manual is non-nil."
+  (interactive "P")
+  (when (org-roam--org-roam-file-p)
+    (org-element-map (org-element-parse-buffer) 'link
+      (lambda (link)
+        (let* ((type (org-element-property :type link))
+               (path (org-element-property :path link))
+               (is-title (if (string= type "roam") t nil))
+               (roam-file (if is-title (org-roam--get-file-from-title path) t)))
+          (unless roam-file
+            (org-roam-link--auto-create-roam-link-file path manual)))))))
+
+(defun org-roam-link--current-buffer-roam-link-titles ()
+  "Return a list of unique roam-link titles in the current buffer."
+  (->> (org-element-map (org-element-parse-buffer) 'link
+         (lambda (link)
+           (let* ((type (org-element-property :type link))
+                  (path (org-element-property :path link))
+                  res)
+             (when (string= type "roam")
+               (cons path res)))))
+       (-flatten)
+       (-distinct)))
+
+(defun org-roam-link--completion (&optional _arg)
+  "Completion for roam-links in `org-mode'.
+ARG is optional prefix supplied through `org-mode'"
```

Comment:
if `arg` is not being used, it can be ignored/renamed to `_` and removed from docstring.